redislock.go

```diff
@@ -98,8 +124,16 @@ func (c *Client) Obtain(ctx context.Context, key string, ttl time.Duration, opt
 	}
 }
 
-func (c *Client) obtain(ctx context.Context, key, value string, ttl time.Duration) (bool, error) {
-	return c.client.SetNX(ctx, key, value, ttl).Result()
+func (c *Client) obtain(ctx context.Context, keys []string, value string, ttl time.Duration) (bool, error) {
+	setArgs := []string{}
```

Comment:
Ideally, prealloc this - `setArgs := make([]string, 0, len(keys) * 2)`.

Or, maybe better call the msetnx script with more usual KEYS=the_keys_array, ARGV=[value, ttl]. Then build the msetnx args in script to save some "traffic".

Doesn't prevent us from merging and then polishing it ourselves.

Reply:
That would be a neat optimization